JDG-2171 Embedded datagrid

karesti opened the pull request:
https://issues.jboss.org/browse/JDG-2171

karesti committed:
JDG-2171 Datagrid Openshift Embedded

oraNod commented on embedded-openshift/README.md:
```diff
@@ -0,0 +1,70 @@
+
+      oc cluster up
+      oc login -u system:admin
+      oc policy add-role-to-user view system:serviceaccount:myproject:default -n myproject
```
Correct me if I'm wrong but I believe you can add the view role to the default service account as the `developer` user because it applies to the current project namespace. Also consider this alternative:
`$ oc policy add-role-to-user view system:serviceaccount:$(oc project -q):default -n $(oc project -q)`

karesti replied:
Ok ! I'm going to test it as you tell me. I took this from kubernetes/openshift example we have in upstream. We might need to update the docs there too.
https://github.com/infinispan/infinispan-simple-tutorials/blob/master/kubernetes/README.adoc

oraNod committed:
JDG-2171 embedded openshift suggested doc edits

JDG-2171 edits to pom file and adding details about namespace to tutorial

JDG-2171 minor edit to pom.xml

JDG-2171 minor edit

galderz commented:
Needs rebasing.

@karesti Even though cache service uses fabric8:run, i don't think it's suited for these demos because it expects to run something that stays up. So once the main completes it keeps trying to restart it. Instead, it's better to make the tutorial be a single shot java process. An example of that can be found [here](https://github.com/infinispan-demos/custom-datagrid-openshift/tree/master/create-cache-permanent).

karesti commented:
@galderz I don't understand why, the other examples in JDG are not one shot. You just start, verify how things are happening and then you stop them when you want. Your suggestion makes more sense to me in the infinispan-simple-tutorials that indeed are most of them one shot super simple examples

galderz commented on embedded-openshift/README.md:
````diff
@@ -0,0 +1,78 @@
+2. Scale the quickstart application up or down.
+
+  ```bash
+  $ oc scale dp jboss-embedded-openshift-quickstart --replicas=<number>
````
Should be `oc scale dc...`

karesti replied:
I let @oraNod change the doc and then we can merge

oraNod replied:
@karesti I've made the change and squashed my commits. @galderz excellent catch!

galderz reviewed (approved):
Small change, the rest is fine. I had missed the loop at the end of the main class, which makes it fine for `fabric8:run`.

karesti commented:
@pruivo feel free to merge this one too, galder approved it

pruivo commented:
integrated! thanks @karesti !
